### PR TITLE
Phala World Shell Parts & Market Updates

### PR DESCRIPTION
**Pull request by HashWarlock**
## Targets
- [x] Add tradable bool field for minting shell part NFTs
- [x] Add tests for marketplace simulated NFT part sell
- [x] Enable `pallet_rmrk_core::send` for user facing function
- [x] Change incubation phase to 8 days
- [x] Disable RMRK Market Pallet Calls

### CODE REVIEW

**Comment by Leechael** [comment]
> Disable RMRK Market Pallet Calls

Is it means RMRK Market Pallet Calls all disabled in this branch? When I try to test the update from this branch, the list extrinsic fail with the message "system.CallFiltered: The origin filter prevent the call to be dispatched".

**Comment by Leechael** [comment]
Based on the example code in #180, I'm try to find out how to trade / send parts NFT, but call `rmrkMarket.list` API failed because it said I not the owner of NFT.

**Comment by HashWarlock** [comment]
> Based on the example code in #180, I'm try to find out how to trade / send parts NFT, but call `rmrkMarket.list` API failed because it said I not the owner of NFT.

This is due to the NFT being nested under a NFT. So the NFT owner is the derived account of the parent NFT. You would need to send the NFT to your `AccountId` before you can `list` the NFT in the marketplace.

**Comment by HashWarlock** [comment]
> > Disable RMRK Market Pallet Calls
> 
> Is it means RMRK Market Pallet Calls all disabled in this branch? When I try to test the update from this branch, the list extrinsic fail with the message "system.CallFiltered: The origin filter prevent the call to be dispatched".

Yes, they are disabled to prevent listing/selling the NFTs until we enable the market functionality after the incubation phase & hatching of Shell NFT.

**Comment by Leechael** [comment]
> > Based on the example code in #180, I'm try to find out how to trade / send parts NFT, but call `rmrkMarket.list` API failed because it said I not the owner of NFT.
> 
> This is due to the NFT being nested under a NFT. So the NFT owner is the derived account of the parent NFT. You would need to send the NFT to your `AccountId` before you can `list` the NFT in the marketplace.

I don't have any experience with that. Can you share the API name or test script with me? I'm looking around rmrkCore and rmrkMarket but not sure which API I should use.

**Comment by Leechael** [comment]
Another small issue I'm found under the test:
1. Lee list his NFT: `rmrkMarket.list(2, 0, 10, null)`.
2. Lee sends the NFT to Joshua: `rmrkCore.send(2, 0, joshua)`.
3. Anyone can continue see the list information (`query.rmrkMarket.listedNfts`), but the owner still Lee and no one can make offer for that.

That more like a bug for RMRK, the `send` extrinstic should unlist a NFT when the send extrinstic successful.

File: pallets/phala-world/src/incubation.rs [hunk 1]
```diff
@@ -709,6 +719,7 @@ where
 	fn can_hatch() -> bool {
 		let now = T::Time::now().as_secs();
 		now > OfficialHatchTime::<T>::get()
+		// true
```
Comment (h4x3rotab) [inline pallets/phala-world/src/incubation.rs:722]: Please remove

**Review by h4x3rotab — COMMENTED** [review]
LGTM

**Review by h4x3rotab — APPROVED** [review]

**Review by Leechael — APPROVED** [review]

**Pull request merged**